app/src/main/java/openfoodfacts/github/scrachx/openfood/views/MainActivity.java

```diff
@@ -61,15 +61,18 @@
 
 public class MainActivity extends BaseActivity implements CustomTabActivityHelper.ConnectionCallback {
 
-    private static final int LOGIN_REQUEST = 1;
     public static final int USER_PROFILE = 100;
+    public static final int LOGIN_ID = 6;
+    private static final int LOGIN_REQUEST = 1;
     private static final long PROFILE_SETTING = 200;
     private static final int CONTRIBUTOR = 300;
     private static final int LOGOUT = 400;
-    public static final int LOGIN_ID = 6;
     private static final long USER_ID = 500;
     private static final int ABOUT = 600;
     private static final int CONTRIBUTE = 700;
+    private static final String CONTRIBUTIONS_SHORTCUT = "openfoodfacts.github.scrachx.openfood.CONTRIBUTIONS";
+    private static final String SCAN_SHORTCUT = "openfoodfacts.github.scrachx.openfood.SCAN";
+    private static final String BARCODE_SHORTCUT = "openfoodfacts.github.scrachx.openfood.BARCODE";
```

Comment:
Could you use `applicationId` instead so this functionality can be also used by the other flavors?

Reply:
I get your issue however the problem is that package name in the `<intent android:targetPackage="" />` in *shortcuts.xml* must be a hardcoded string.
https://stackoverflow.com/questions/40361296/static-android-shortcuts-for-multiple-flavors

Reply:
We might want use https://github.com/Zellius/android-shortcut-gradle-plugin to generate them

Reply:
Okay I will try to use this plugin. Will push if successfully done.